packages/ember-metal/lib/properties.js

```diff
@@ -83,61 +83,71 @@ export function DEFAULT_GETTER_FUNCTION(name) {
   @param {*} [data] something other than a descriptor, that will
     become the explicit value of this property.
 */
-export function defineProperty(obj, keyName, desc, data, meta) {
-  var possibleDesc, existingDesc, watching, value;
+export function defineProperty(obj, keyName, desc, data/*, meta*/) {
+  let meta = arguments[4] || metaFor(obj);
```

Comment:
a meta may not exists here yet, we should be careful to not create on if it does not need to exist.